website/server/models/group.js

```diff
@@ -1597,7 +1593,16 @@ schema.methods.syncTask = async function groupSyncTask (taskToSync, user) {
   // add tag if missing
   if (matchingTask.tags.indexOf(group._id) === -1) matchingTask.tags.push(group._id);
 
-  toSave.push(matchingTask.save(), taskToSync.save(), user.save());
+  toSave.push(matchingTask.save(), user.save());
+  return Promise.all(toSave);
+};
+
+schema.methods.linkTask = async function groupLinkTask (linkingTask, user) {
```

Comment:
we should start to split up this file, it's really huge, might be a good occasion to start by moving these tasks related methods

Reply:
What's the difference between link and sync task? Can we add some comments explaining how they work and how they're supposed to be used?